samples/v2/index-create.js

```diff
@@ -0,0 +1,77 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// sample-metadata:
+//  title: Creates a new index
+//  usage: node createIndex <INSTANCE_ID> <DATABASE_ID> <PROJECT_ID>
+
+'use strict';
+
+function main(
+  instanceId = 'my-instance',
+  databaseId = 'my-database',
+  projectId = 'my-project-id'
+) {
+  // [START spanner_create_index]
+  /**
+   * TODO(developer): Uncomment these variables before running the sample.
+   */
+  // const instanceId = 'my-instance';
+  // const databaseId = 'my-database';
+  // const projectId = 'my-project-id';
+
+  // Imports the Google Cloud client library
+  const {Spanner} = require('@google-cloud/spanner');
+
+  // creates a client
+  const spanner = new Spanner({
+    projectId: projectId,
+  });
+
+  const databaseAdminClient = spanner.getDatabaseAdminClient();
+
+  async function createIndex() {
+    const request = ['CREATE INDEX AlbumsByAlbumTitle ON Albums(AlbumTitle)'];
+
+    // Creates a new index in the database
+    try {
+      const [operation] = await databaseAdminClient.updateDatabaseDdl({
+        database: databaseAdminClient.databasePath(
+          projectId,
+          instanceId,
+          databaseId
+        ),
+        statements: request,
+      });
+
+      console.log('Waiting for operation to complete...');
+      await operation.promise();
+
+      console.log('Added the AlbumsByAlbumTitle index.');
+    } catch (err) {
+      console.error('ERROR:', err);
+    } finally {
+      // Close the spanner client when finished.
+      // The databaseAdminClient does not require explicit closure. The closure of the Spanner client will automatically encompass the closure of the databaseAdminClient.
+      spanner.close();
```

Comment:
I assume that the reason that we are sometimes closing this client and sometimes not in all of these samples, is that the previous samples also had the same inconsistencies. Can we have a follow-up PR that makes this consistent across all samples?

Reply:
Sure